File: LvlUpBack-main/src/main/java/com/levelup/gaming/models/Order.java
package com.levelup.gaming.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "orders")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Order {
    @Id
    private String id;
    private String userId;

    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JoinColumn(name = "order_id")
    private List<OrderItem> items = new ArrayList<>();

    @Embedded
    private Address shippingAddress;

    private String paymentMethod;
    private double totalPrice;
    private double shippingPrice;
    private boolean isPaid;
    private String status;
    private LocalDateTime createdAt;
    private LocalDateTime paidAt;
    private LocalDateTime deliveredAt;
}

File: LvlUpFront-main/src/types/Order.ts


export type OrderStatus = 'Pendiente' | 'Procesando' | 'Enviado' | 'Entregado' | 'Cancelado';

export interface OrderItem {
    id?: number;
    productId?: string;
    name: string;
    price: number;
    qty: number;
    quantity?: number; 
    image?: string;
}

export interface ShippingAddress {
    street: string;
    city: string;
    region: string;
}

export interface Order {
    id: string;
    userId: string;
    userRut?: string; 
    items: OrderItem[];
    shippingAddress: ShippingAddress;
    paymentMethod: string;          
    totalPrice: number;
    shippingPrice: number;          
    isPaid: boolean;                
    status: OrderStatus;
    createdAt: string;
    paidAt?: string;                
    deliveredAt?: string;           
}

export interface OrderUpdatePayload {
    status: OrderStatus;
}
